models/cohortdata.go

```diff
@@ -70,6 +70,29 @@ func (h CohortData) RetrieveDataBySourceIdAndCohortIdAndConceptIdsOrderedByPerso
 	return cohortData, meta_result.Error
 }
 
+func (h CohortData) RetrieveHistogramDataBySourceIdAndCohortIdAndConceptIdsAndCohortPairs(sourceId int, cohortDefinitionId int, histogramConceptId int64, filterConceptIds []int64, filterCohortPairs []utils.CustomDichotomousVariableDef) ([]*PersonConceptAndValue, error) {
+	log.Printf(">> Using inner join impl. for large cohorts")
+	var dataSourceModel = new(Source)
+	omopDataSource := dataSourceModel.GetDataSource(sourceId, Omop)
+
+	resultsDataSource := dataSourceModel.GetDataSource(sourceId, Results)
+
+	// get the observations for the subjects and the concepts, to build up the data rows to return:
+	var histogramValueFieldName = "observation.value_as_number"
+	var cohortData []*PersonConceptAndValue
+	query := omopDataSource.Db.Model(&Observation{}).
+		Select("distinct(observation.person_id), observation.observation_concept_id as concept_id, observation.value_as_string as concept_value_as_string, observation.value_as_number as concept_value_as_number, observation.value_as_concept_id as concept_value_as_concept_id").
+		Joins("INNER JOIN "+resultsDataSource.Schema+".cohort as cohort ON cohort.subject_id = observation.person_id").
```

Comment:
let's keep only `observation.value_as_number` given this method is for numeric values only

Reply:
can i keep the distinct(observation.person_id) and observation.value_as_concept_id as concept_value_as_concept_id for testing?

Reply:
ah, I meant: from all the value_as_ fields, keep only the value_as_number :)